state/action.go

```diff
@@ -449,20 +458,23 @@ func (a *action) removeAndLogBuildTxn(finalStatus ActionStatus, results map[stri
 					}
 				}
 				updateOperationOp = &txn.Op{
-					C:      operationsC,
-					Id:     a.st.docID(parentOperation.Id()),
-					Assert: assertNotComplete,
+					C:  operationsC,
+					Id: a.st.docID(parentOperation.Id()),
+					Assert: append(assertNotComplete,
+						bson.DocElem{"complete-task-count", bson.D{{"$eq", expectedTaskCount - 1}}}),
 					Update: bson.D{{"$set", bson.D{
 						{"status", finalOperationStatus},
 						{"completed", completedTime},
-						{"complete-task-count", numComplete + 1},
+						{"complete-task-count", expectedTaskCount},
 					}}},
 				}
 			} else {
 				updateOperationOp = &txn.Op{
-					C:      operationsC,
-					Id:     a.st.docID(parentOperation.Id()),
-					Assert: bson.D{{"complete-task-count", bson.D{{"$lt", len(tasks) - 1}}}},
+					C:  operationsC,
+					Id: a.st.docID(parentOperation.Id()),
+					Assert: append(assertNotComplete,
+						bson.DocElem{"complete-task-count",
```

Comment:
```suggestion
						bson.DocElem{"spawned-task-count",
```

Reply:
When updating the operation because an action completed, though not completing the operation, the complete task count needs to always be less than the spawned task count minus 1.  That allows for the operation to be completed.

Reply:
Doh! I should have read that part of the code more carefully; I see now. Please ignore my other comments below!